lib/Models/ItemSearchProviders/Types.ts

```diff
@@ -1,7 +1,7 @@
 export type IndexBase<QueryType> = {
   type: IndexType;
   load(indexRootUrl: string, valueHint: QueryType): Promise<void>;
-  search(query: QueryType): Promise<Set<number>>;
+  search(query: QueryType, queryOptions: any): Promise<Set<number>>;
```

Comment:
Rather than passing a queryOptions to every index type, including the ones that don't use it, I would make it optional and remove it as a `search` parameter in the things that aren't using it. It's not a huge deal, but the code is slightly clearer that way. What do you think?

Reply:
Good point!